packages/cache-utils/src/hash.js

```diff
@@ -1,135 +1,32 @@
 
-const pStat = promisify(stat)
-const pReadFile = promisify(readFile)
-
-// Compute the hash of a file's contents.
-// Use SHA256 on the contents. Also hashes some file metadata: file path, file
-// type, permissions, uid/gid.
-const getHash = async function({ srcPath, move, digests, base }) {
+// Caching a big directory like `node_modules` is slow. However those can
+// sometime be represented by a digest file such as `package-lock.json`. If this
+// has not changed, we don't need to save cache again.
+const getHash = async function(digests, move) {
   // Moving files is faster than computing hashes
-  if (move) {
-    return ''
-  }
-
-  const { digestPath, fileStat } = await findDigest(srcPath, digests)
-
-  if (fileStat.isDirectory()) {
-    return hashDir(digestPath, fileStat, base)
-  }
-
-  return hashFile(digestPath, fileStat, base)
-}
-
-// Computing a big directory like `node_modules` is slow. However those can
-// sometime be represented by a digest file such as `package-lock.json` which
-// is much smaller and fast to hash. We allow specifying those using the
-// `digests` option.
-const findDigest = async function(srcPath, digests) {
-  const fileStats = await Promise.all([...digests, srcPath].map(getStat))
-  return fileStats.find(Boolean)
-}
-
-const getStat = async function(digestPath) {
-  try {
-    const fileStat = await pStat(digestPath)
-    return { digestPath, fileStat }
-  } catch (error) {
+  if (move || digests.length === 0) {
     return
   }
-}
-
-// Hash a directory recursively
-const hashDir = async function(dirPath, fileStat, base) {
-  const files = await readdirp.promise(dirPath, { fileFilter, alwaysStat: true })
-  const dirHashInfo = getHashInfo(dirPath, fileStat, base)
-  const hashInfos = await pMap(files, ({ fullPath, stats }) => getFileInfo(fullPath, stats, base), {
-    concurrency: MAX_CONCURRENCY,
-  })
-  const hash = await computeHash([dirHashInfo, ...hashInfos])
-  return hash
-}
-
-// We do not copy temporary files like `*~` nor hash them
-const fileFilter = function({ basename }) {
-  return junk.not(basename)
-}
-
-// Hash a regular file
-const hashFile = async function(filePath, fileStat, base) {
-  const hashInfo = await getFileInfo(filePath, fileStat, base)
-  const hash = await computeHash([hashInfo])
-  return hash
-}
-
-// Avoid memory crashes due to too high parallelism.
-// This is mostly CPU-bound (hashing), so higher values actually make it slower.
-const MAX_CONCURRENCY = cpus().length
-
-const getFileInfo = async function(filePath, stat, base) {
-  const content = await getContent(filePath, stat)
-  const hashInfo = getHashInfo(filePath, stat, base, content)
-  return hashInfo
-}
-
-// If a file is too big, we stream it and hash it first so it does not take up
-// memory. Otherwise files with lots of big files can end up taking the whole
-// memory and crash the process.
-const getContent = function(filePath, { size }) {
-  if (size < MAX_HASH_CONTENT) {
```

Comment:
We could just remove this condition and hash all files and keep the hash in memory instead of the files content, to avoid holding many files in memory when dealing with directories with a lot of small/many files.

Reply:
Yes you're correct! We should do this if we introduce that code again.